docs/src/main/asciidoc/se/injection.adoc

```diff
@@ -0,0 +1,850 @@
+However, keep in mind that usage of this annotation doesn’t add the ability to intercept constructor calls.
+To enable interception, this annotation must be present on the class that the factory produces.
+While it is not required on interfaces, it will still work correctly if applied there.
+
+If you need to enable interception for classes using delegation, you should make sure about the following:
+
+- The class must have accessible no-arg constructor (at least package local) and mustn’t be final
+- The constructor should have no side effects, as the instance will act only as a wrapper for the delegate
+- All invoked methods must be accessible (at least package local) and non-final
+
+Therefore, using link:{service-registry-base-url}/io/helidon/service/registry/Interception.Delegate.html[`@Interception.Delegate`]
+on the classes can be considered as acknowledgement of the rules above.
+
+[source,java]
+.Delegate used on the class
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_3, indent=0]
+----
+
+=== ExternalDelegate annotation
+The link:{service-registry-base-url}/io/helidon/service/registry/Interception.ExternalDelegate.html[`@Interception.ExternalDelegate`]
+annotation works similarly to
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.Delegate.html[`@Interception.Delegate`].
+However, the key difference is that
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.ExternalDelegate.html[`@Interception.ExternalDelegate`]
+is designed for classes that you don’t have control over.
+This means it allows you to apply the interception mechanism even to third-party classes.
+
+It is not required to apply this annotation on the interfaces, however, it needs to be present for classes.
+
+Let's make the same `@Traced` annotation and Interceptor as in the previous examples
+
+[source,java]
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_1, indent=0]
+----
+
+In this example, we can’t change the class we’re intercepting in any way.
+Let's assume we have this external class, and we need to add an interception mechanism to it.
+
+[source,java]
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_4, indent=0]
+----
+
+Now we need to apply the
+link:{service-registry-base-url}/io/helidon/service/registry/Interception.ExternalDelegate.html[`@Interception.ExternalDelegate`]
+annotation on the factory class. Once this is done, interception will work as expected
+
+[source,java]
+.ExternalDelegate annotation used on the factory
+----
+include::{sourcedir}/se/inject/InterceptorDelegateExample.java[tag=snippet_5, indent=0]
+----
+
+== Events
+Events enable in-application communication between services by providing a mechanism to emit events and register consumers to handle them.
+
+A single event can be delivered to zero or more consumers.
+
+Key Terminology:
+
+- *Event Object* – Any object that is sent as an event.
```

Comment:
These should be links to the chapters below (`Event Object`, `Event Emitter` etc.)